Document new way to not build LLVM (#878)
rust-lang/rustc-dev-guide: src/building/suggested.md

You can then use that rust2 folder as a separate workspace for modifying
and building `rustc`!

## Skipping LLVM Build

By default, LLVM is built from source, and that takes significant amount of
time. One way to avoid that is to add this to `config.toml`:

```toml
[llvm]
download-ci-llvm = true
```

> Downloading LLVM from CI is still experimental though, and might not be
> available on all platforms. Otherwise, we'd make it a default!

Reviewer: ```suggestion
Downloading LLVM from CI is still experimental though, and is currently only
available on Linux. Otherwise, we'd make it a default!
```

Author: I somewhat intentionally phrased this vaguely, so that it does not bitrot, but I am fine with either wording.

Reviewer 2: I think we can keep it as is personally.

> available on all platforms. Otherwise, we'd make it a default!

Reviewer 3: This does not feel like the right place to indicate planned policy/default, here referring to "we'd make it a default".

Author: This also was done in the interest of future compatibility. The thing is in flux, so I find it useful:

* to document it early, even if incomplete
* to be clear that there might be changes, and that the docs might be out of date
* to not be super strict with keeping the docs up to date.

Reviewer 3: I was only referring to "Other, we'd make it a default". First sentence is all good.
